### DISCUSSION TITLE
Integration with orgajs

### DISCUSSION THREAD

**Discussion by xiaoxinghu** [system-ui/theme-ui#1925] [Ideas] [answered]
Hi, I am the author of [orgajs](https://github.com/orgapp/orgajs). orgajs is a [org-mode](https://orgmode.org) syntax parser heavily influenced by mdx. They basically work the same way. I am thinking about making a package like [@theme-ui/mdx](https://github.com/system-ui/theme-ui/tree/develop/packages/mdx) to have better integration with theme-ui. It should be mostly a direct copy of the package with trivial changes. I just want to hear your options.
- should I just copy & paste majority of the code from @theme-ui/mdx, and replace mdx with orgajs? do you have a better way of doing this?
- should I contribute to theme-ui repo or add that package to orgajs? so `@theme-ui/orga` vs. `@orgajs/theme-ui`.

Thanks!

**Comment by lachlanjc** [comment, marked as answer]
Appreciate the interest & patience!

> should I just copy & paste majority of the code from @theme-ui/mdx, and replace mdx with orgajs? do you have a better way of doing this?

I don't think there is a better way. That said, you could certainly simplify that package a lot—you wouldn't want/need several parts of Theme UI which we just happen to implement in that package.

> should I contribute to theme-ui repo or add that package to orgajs? so @theme-ui/orga vs. @orgajs/theme-ui.

Definitely go for the latter, as you'll be a better steward & know how to evolve it more effectively.

Best of luck & please let me know if you have questions!
